src/libmatrix/util.cc

```diff
@@ -244,7 +244,7 @@ Util::appname_from_path(const std::string& path)
 std::istream *
 Util::get_resource(const std::string &path)
 {
-    std::ifstream *ifs = new std::ifstream(path.c_str());
+    std::ifstream *ifs = new std::ifstream(path.c_str(), std::ios::in | std::ios::binary);
```

Comment:
I noticed this too. But looking at the api:

explicit basic_ifstream( const char* filename,
                std::ios_base::openmode mode = ios_base::in );

(from https://en.cppreference.com/w/cpp/io/basic_ifstream/basic_ifstream )

it looks like if I override the second argument, I must explicitly "or" in ios::in. Or else we would override the open mode without in being implicitly defined.

Reply:
The default value of the ifstream constructor mode argument is unfortunately misleading in this way. From the same cppreference page:

> (2-3) First, performs the same steps as the default constructor, then associates the stream with a file by calling rdbuf()->open(filename, **mode | std::ios_base::in**)

So, whatever mode we pass in, it's combined with std::ios_base::in anyway. This makes sense, since otherwise we could break the main contract of ifstream, and end up with an input stream we couldn't read from. In any case, as I mentoned, this is just a nit, I brought this up mostly because I found it to be an interesting detail :)

Reply:
Ah yes, good point. I can maybe fix that at some point in a follow-up patch.